lisa/sut_orchestrator/hyperv/serial_console_logger.ps1

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+param (
+    [Parameter(Mandatory=$true)][string]$PipeName,
+    [Parameter(Mandatory=$true)][string]$OutFile,
+    [Parameter(Mandatory=$false)][int]$PipeReconnectMillis = 12000
+)
+
+Write-Host "PID: $PID"
+Write-Host "PipeName: $PipeName"
+Write-Host "OutFile: $OutFile"
+
+$oStream = New-Object System.IO.FileStream $OutFile, 'OpenOrCreate', 'Write', 'Read'
+$pClient = New-Object System.IO.Pipes.NamedPipeClientStream(".", $PipeName)
+$reconnect = $false
+
+try {
+    while (1) {
+        $timeout = 60000
+        if ($reconnect) {
+            $timeout = $PipeReconnectMillis
+        }
+
+        $pClient.Connect($timeout)
+        if (!$pClient.IsConnected) {
+            Write-Host "Failed to connect to pipe. Exiting..."
+            exit
+        }
+
+        Write-Host "Connected to pipe"
+        $pClient.CopyTo($oStream)
+
+        Write-Host "Disconnected from pipe. Reconnecting..."
+        $reconnect = $true
```

Comment:
When will it be disconnected, if above is block operation?

Reply:
It gets briefly disconnected when the VM is rebooted.
I will add some comments to this file.

Reply:
Added some comments.